files/build_templates/docker_image_ctl.j2

```diff
@@ -146,6 +146,8 @@ start() {
         -e PRM_SNIFFER \
         -e PRM_SNIFFER_FILE_PATH \
         -v /var/log/mellanox/sniffer:/var/log/mellanox/sniffer:rw \
+        -v sx_api:/tmp \
+        -v /dev/shm:/dev/shm:rw \
```

Comment:
```
{%- if sonic_asic_platform == "mellanox" %}
{%- if docker_container_name == "syncd" %}
        -e SX_SNIFFER_ENABLE \
        -e SX_SNIFFER_TARGET \
        -e PRM_SNIFFER \
        -e PRM_SNIFFER_FILE_PATH \
        -v /var/log/mellanox/sniffer:/var/log/mellanox/sniffer:rw \
        -v sx_api:/tmp \
        -v /dev/shm:/dev/shm:rw \
{%- else %}
        --tmpfs /tmp \
{%- endif %}
{%- if docker_container_name == "pmon" %}
        -v /var/run/hw-management:/var/run/hw-management:rw \
{%- endif %}
{%- endif %}
```

Reply:
What is the benefit to add 'tmpfs'? Why syncd is an exception for it?

---
In reply to: [288094235](https://github.com/Azure/sonic-buildimage/pull/2951#discussion_r288094235) [](ancestors = 288094235)

Reply:
I see the point, but if you could make one less exception, that will be better.

Reply:
To be more constructive, I think the path `/tmp/sx_api` is weird. If Mellanox plan to share it across processes or across containers, you'd better config it to a proper path. `/tmp` is not used for this purpose.

---
In reply to: [288366517](https://github.com/Azure/sonic-buildimage/pull/2951#discussion_r288366517) [](ancestors = 288366517)

Reply:
> To be more constructive, I think the path `/tmp/sx_api` is weird. If Mellanox plan to share it across processes or across containers, you'd better config it to a proper path. `/tmp` is not used for this purpose.
> 
> If you could move it, this exception is not existing.
> 
> In reply to: [288366517](https://github.com/Azure/sonic-buildimage/pull/2951#discussion_r288366517) [](ancestors = 288366517)

Discussing internally with SDK team with the '/tmp' folder, once have conclusion will update.

Reply:
@qiluo-msft as discussed, we will remove the --tmpfs exception after socket path change in new SDK version, TODO added.